fix(how-to): links to spec

scop opened the pull request:
HTML rendering does not appear to remove `.md` to "external" doc links.

lehors commented:
Sorry, I beat you to it. PR #1282 addressed the problem. Note that this only occurs with links that have a fragment identifier. I don't know for sure but it looks to me like a bug in Jekyll.
Thanks though.

scop commented:
It does not seem to be limited to links with fragments. I checked every instance that I fixed in my version here before fixing them, and they were all broken similarly. The ones that seem to remain unaddressed are:

- The "SLSA Levels" link in https://slsa.dev/how-to/how-to-orgs#choosing-your-slsa-level
- The "Verifying Artifacts" link at end of https://slsa.dev/how-to/how-to-orgs#for-software-consumers

I've rebased & reworked my branch here to address those.

scop commented:
Hm, updating my branch did not seem to affect what is shown here, so I opened a new one with the remaining fixes: https://github.com/slsa-framework/slsa/pull/1286
